### PR TITLE
added 3 months cap for since and some error handling string fromattin…

### PR DESCRIPTION
**Pull request by galiri**
…g tidy

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,7 +52,8 @@ Read notifications:
 curl http://localhost:8080/lists/notifications?since=$date
 ```
 
-Where `$date` is a date in RFC3339 format which is within the last 3 months. For an example date, simply hit the `/lists/notifications` endpoint with no since parameter.
+Where `$date` is a date in RFC3339 format which is within the last 3 months.  simply hit the `/lists/notifications` endpoint with no since parameter.
+e.g. since=2016-11-02T12:41:47.4692365ZFor an example date.
```
Comment (khatton-ft) [inline README.md]: This looks like something got scrambled.

File: resources/read_test.go [hunk 2]
```diff
@@ -121,22 +121,35 @@ func Test400JunkSinceDate(t *testing.T) {
...
+	ReadNotifications(testMapper, testLinkGenerator, mockDb, 1000)(w, req)
 
 	assert.Equal(t, 400, w.Code, "The since date was garbage! Should be 400!")
 
 	mockDb.AssertNotCalled(t, "Open")
 	t.Log("Recorded 400 response as expected.")
 }
 
+func Test400SinceDateTooEarly(t *testing.T) {
+	req, _ := http.NewRequest("GET", "http://nothing/at/all", nil)
```
Comment (Nesze) [inline resources/read_test.go]: I think this test passes because there is no since param specified

File: resources/read.go [hunk 3]
```diff
@@ -17,26 +17,31 @@ type msg struct {
...
 		if param == "" {
-			logrus.Warn("User didn't provide since date.")
+			logrus.Info("User didn't provide since date.")
 			writeMessage(400, sinceMessage(), w)
 			return
 		}
 
-		since, err := time.Parse(time.RFC3339Nano, param)
+		since, err := time.Parse(time.RFC3339Nano, param)	
+    	if(since.Before(time.Now().UTC().AddDate(0, -maxSinceInterval, 0))){
```
Comment (peteclark-ft) [inline resources/read.go]: The formatting here isn't right - I recommend using Atom with the go-plus plugin, it'll automatically correct this on save.

File: main.go [hunk 4]
```diff
@@ -40,10 +40,15 @@ func main() {
 			Usage: "The max age for content records in varnish in seconds.",
 			Value: 10,
 		}),
+		altsrc.NewIntFlag(cli.IntFlag{
+			Name:  "max-since-interval",
+			Usage: "The maximum time interval clients are allowed to query for notifications in months.",
+			Value: 3,
```
Comment (peteclark-ft) [inline main.go]: Should this be in days for more flexibility?

File: resources/read_test.go [hunk 5]
```diff
@@ -121,22 +121,35 @@ func Test400JunkSinceDate(t *testing.T) {
...
+	w := httptest.NewRecorder()
+
+	mockDb := new(MockDB)
+	ReadNotifications(testMapper, testLinkGenerator, mockDb, 3)(w, req)
+
+	assert.Equal(t, 400, w.Code, "Sice date too early, should be 400!")
```
Comment (peteclark-ft) [inline resources/read_test.go]: Since*

File: resources/read.go [hunk 6]
```diff
@@ -17,26 +17,31 @@ type msg struct {
...
+    		logrus.Infof("User provided since date before query cap date, since= [%v].", since.Format(time.RFC3339Nano))
+			writeMessage(400, fmt.Sprintf("since date must be within the last %d months.", maxSinceInterval), w)
```
Comment (peteclark-ft) [inline resources/read.go]: Nitpicky, but this is a customer facing error message: should be `Since`

File: resources/read_test.go (hunk 2 above)
Comment (peteclark-ft) [inline resources/read_test.go]: +1

File: README.md (hunk 1 above)
Comment (galiri) [inline README.md]: unscrambled

File: resources/read_test.go (hunk 2 above)
Comment (galiri) [inline resources/read_test.go]: done

**Pull request merged**